src/services/User.service.js

```diff
@@ -11,35 +10,119 @@ class UserServices {
    * @returns {object} user data from database
    */
   async findUserByEmail(email) {
-    const user = await db.user.findOne({
-      where: { email }
-    });
+    const user = await db.user.findOne({ where: { email } });
     if (!user) return null;
+    const lastLogin = new Date().toISOString();
+    await db.user.update({ lastLogin }, { where: { email } });
     return user;
   }
 
   /**
-   * setUserRole - used to update a user's role by the email provided
-   * @param {object} body - request object posted to route
-   * @returns {object} updated user object
+   * @name GetUserById
+   * @description Interacts with model to find a single user
+   * @param { string } id the user's id
+   * @returns {object} return the user's data
    */
-  async setUserRole(body) {
-    try {
-      const updateUser = await db.user.update(
+  async getUserById(id) {
+    const userData = await db.user.findOne({
+      where: { id },
+      attributes: [
+        'firstName',
+        'lastName',
+        'email',
+        'isVerified',
+        'birthDate',
+        'residenceAddress',
+        'preferredLanguage',
+        'preferredCurrency',
+        'department',
+        'lineManagerId',
+        'gender',
+        'lastLogin',
+        'role',
+        'phoneNumber'
+      ],
+      include: ['LineManager']
+    });
+
+    return userData;
+  }
+
+  /**
+   * @name updateUserInfoByEmail
+   * @description Interacts with model to find a single user
+   * @param { object } attributes the user attribute to update
+   * @param { string } email$ the user's email
+   * @returns {object} return the user's data
+   */
+  async updateUserInfoByEmail(attributes, email$) {
+    const {
+      firstName, lastName, email, birthDate, preferredLanguage, preferredCurrency, residenceAddress,
+      gender, department, lineManagerId, phoneNumber
+    } = attributes;
+
+    const currentUserId = (await db.user.findOne({
+      where: { email: email$ },
+      attributes: ['id']
+    })).dataValues.id;
+
+    const isLineManagerValid = (await db.user.findOne({ where: { id: lineManagerId } })).dataValues.role === 'manager';
+
```

Comment:
Sorry, I noticed this while testing that, if user updates with the profile with a line manager that does not exist it throws a `500` instead of giving out a specific message can you please handle that?? @gildniy